crates/services/gas_price_service/src/v1/da_source_service/block_committer_costs.rs

```diff
@@ -142,54 +142,315 @@ impl BlockCommitterHttpApi {
+
+        // when
+        let actual =
+            rt.block_on(async { block_committer.get_latest_costs().await.unwrap() });
+
+        // then
+        assert_eq!(actual, None);
+    }
+
+    #[test]
+    fn get_latest_costs__when_url_is_some__then_returns_expected_costs() {
+        let rt = tokio::runtime::Runtime::new().unwrap();
+        let mut mock = FakeServer::new();
+        let url = mock.url();
+
+        // given
+        let block_committer = BlockCommitterHttpApi::new(Some(url));
+        let not_expected = RawDaBlockCosts {
+            bundle_id: 1,
+            start_height: 1,
+            end_height: 10,
+            da_block_height: 1u64.into(),
+            cost_wei: 1,
+            size_bytes: 1,
+        };
+        mock.add_response(not_expected);
+        let expected = RawDaBlockCosts {
+            bundle_id: 2,
+            start_height: 11,
+            end_height: 20,
+            da_block_height: 2u64.into(),
+            cost_wei: 2,
+            size_bytes: 2,
+        };
+        mock.add_response(expected.clone());
+
+        // when
+        let actual =
+            rt.block_on(async { block_committer.get_latest_costs().await.unwrap() });
+
+        // then
+        assert_eq!(actual, Some(expected));
+    }
+}
+#[cfg(any(test, feature = "test-helpers"))]
+pub mod fake_server {
+    use crate::v1::da_source_service::block_committer_costs::RawDaBlockCosts;
+    use mockito::Matcher::Any;
+    use std::{
+        collections::HashMap,
+        sync::{
+            Arc,
+            Mutex,
+        },
+    };
+
+    pub struct FakeServer {
+        server: mockito::ServerGuard,
+        responses: Arc<Mutex<Vec<RawDaBlockCosts>>>,
+    }
+
+    impl FakeServer {
+        pub fn new() -> Self {
+            let server = mockito::Server::new();
+            let responses = Arc::new(Mutex::new(Vec::new()));
+            let mut fake = Self { server, responses };
+            fake.init();
+            fake
+        }
+
+        #[allow(unused_variables)]
+        pub fn init(&mut self) {
+            let shared_responses = self.responses.clone();
+            self.server
+                .mock("GET", Any)
+                .with_status(201)
+                .with_body_from_request(move |request| {
+                    // take the requested number and return the corresponding response from the `responses` hashmap
+                    let path = request.path_and_query();
+                    tracing::info!("Path: {:?}", path);
+                    let query = path.split("variant=").last().unwrap();
+                    tracing::info!("Query: {:?}", query);
+                    let mut values = query.split('&');
+                    let variant = values.next().unwrap();
+                    tracing::info!("Variant: {:?}", variant);
+                    match variant {
+                        // Latest: http://localhost:8080/v1/costs?variant=latest&limit=5
+                        // We don't support `limit` yet!!!!
```

Comment:
Do we need it?

Reply:
maybe in the future, we could collect `n` latest da costs, depending on how far back the gas price db is out of sync with enchain db

Reply:
No. Not for our current code. In the future if we support larger values than 1, then we can add it to our mock as well.